### PR TITLE
update dev app to test obo rt

### PR DESCRIPTION
**Pull request by jennyf19**

### CODE REVIEW

File: tests/WebAppCallsWebApiCallsGraph/TodoListService/Controllers/Callback.cs [hunk 1]
```diff
@@ -0,0 +1,69 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
```
Comment (jmprieur) [inline tests/WebAppCallsWebApiCallsGraph/TodoListService/Controllers/Callback.cs:1]: You might want to rename this file to CallbackController.cs

File: tests/WebAppCallsWebApiCallsGraph/TodoListService/Controllers/Callback.cs [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+namespace TodoListService.Controllers
+{
+    [Authorize]
+    [Route("api/[controller]")]
+    //[RequiredScope("access_as_user")] 
+    public class CallbackController : Controller
+    {
+        private readonly ITokenAcquisition _tokenAcquisition; // do not remove
+        // The Web API will only accept tokens 1) for users, and 2) having the access_as_user scope for this API
```
Comment (jmprieur) [inline tests/WebAppCallsWebApiCallsGraph/TodoListService/Controllers/Callback.cs:28]: We can remove this comment. This controller allows anonymous.

File: tests/WebAppCallsWebApiCallsGraph/TodoListService/Controllers/LongRunningProcessContext.cs [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+{
+    public class LongRunningProcessContext : IDisposable
+    {
+        JwtSecurityToken _savedToken;
+        HttpContext _httpContext;
+
+        internal LongRunningProcessContext(HttpContext httpContext, JwtSecurityToken tokenToUse)
+        {
+            _httpContext = httpContext;
+            _savedToken = httpContext.Items["JwtSecurityTokenUsedToCallWebAPI"] as JwtSecurityToken;
```
Comment (jmprieur) [inline tests/WebAppCallsWebApiCallsGraph/TodoListService/Controllers/LongRunningProcessContext.cs:15]: Too bad we cannot use (as they are internal).
https://github.com/AzureAD/microsoft-identity-web/blob/800b833ddf788032ae885807dbbceb7581db8852/src/Microsoft.Identity.Web/HttpContextExtensions.cs#L17

and

https://github.com/AzureAD/microsoft-identity-web/blob/800b833ddf788032ae885807dbbceb7581db8852/src/Microsoft.Identity.Web/HttpContextExtensions.cs#L31

We also have an issue where the customers would like us to Store the incoming token somewhere.

File: tests/WebAppCallsWebApiCallsGraph/TodoListService/Controllers/LongRunningProcessContextFactory.cs [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        /// Get a key associated with the current incoming token
+        /// </summary>
+        /// <param name="httpContext">Http context in which the controller action is running</param>
+        /// <returns>A unique string repesenting the incoming token to the web API. This
+        /// key will be used in the future to retrieve the incoming token even if it has expired therefore
+        /// enabling getting an OBO token.</returns>
+        public string CreateKey(HttpContext httpContext)
+        {
+            JwtSecurityToken token = httpContext.Items["JwtSecurityTokenUsedToCallWebAPI"] as JwtSecurityToken;
+            string key = "1";
```
Comment (jmprieur) [inline tests/WebAppCallsWebApiCallsGraph/TodoListService/Controllers/LongRunningProcessContextFactory.cs:33]: you might want to have something else here.
"1" was nice to test, but probably have one per user would be nice?

Maybe key = `httpContext.GetMsalId()` or `Guid.NewGuid().string()`

**Review by jmprieur — APPROVED** [review]
LGTM.
Thanks!

File: tests/WebAppCallsWebApiCallsGraph/TodoListService/Controllers/Callback.cs (hunk 1 above)
Comment (jennyf19) [reply tests/WebAppCallsWebApiCallsGraph/TodoListService/Controllers/Callback.cs]: done.

File: tests/WebAppCallsWebApiCallsGraph/TodoListService/Controllers/Callback.cs (hunk 2 above)
Comment (jennyf19) [reply tests/WebAppCallsWebApiCallsGraph/TodoListService/Controllers/Callback.cs]: done.

File: tests/WebAppCallsWebApiCallsGraph/TodoListService/Controllers/LongRunningProcessContext.cs (hunk 3 above)
Comment (jennyf19) [reply tests/WebAppCallsWebApiCallsGraph/TodoListService/Controllers/LongRunningProcessContext.cs:15]: agree...should we open an issue?

File: tests/WebAppCallsWebApiCallsGraph/TodoListService/Controllers/LongRunningProcessContextFactory.cs (hunk 4 above)
Comment (jennyf19) [reply tests/WebAppCallsWebApiCallsGraph/TodoListService/Controllers/LongRunningProcessContextFactory.cs]: yep. thanks.

**Pull request merged**